src/components/common/DataTable.vue

```diff
@@ -79,38 +79,48 @@ export interface Item {
 export default class DataTable extends Vue {
   @Prop() fields!: ViewTableField[];
   @Prop() items!: Item[];
-  @Prop() filter!: string;
-  @Prop() filterBy!: string;
-  @Prop() defaultSort!: string;
+  @Prop() filter?: string;
+  @Prop() filterBy?: string;
+  @Prop() defaultSort?: string;
   @Prop({ default: "desc" }) defaultOrder!: "desc" | "asc";
   @Prop({ default: 10 }) perPage!: number;
   @Prop({ default: false }) hidePagination!: boolean;
   @Prop() filterFunction?: Function;
 
-  sortBy: string = this.defaultSort;
+  sortBy: string = ''
   descOrder: boolean = this.defaultOrder === "desc";
   currentPage = 1;
 
+  created() {
+    this.sortBy = this.defaultSort ? this.defaultSort : '';
+  }
+
   get modifiedItems() {
     let filtered = [];
     const items = this.items;
     const filter = this.filter;
     const filterBy = this.filterBy;
     const filterFunction = this.filterFunction;
+    const sortBy = this.sortBy
 
     if (filterFunction !== undefined) {
       filtered = items.filter((t: any) => filterFunction(t, filter));
-    } else {
+    } else if (filter && filterBy) {
       filtered = items.filter(
         (t: any) =>
           t[filterBy] &&
           t[filterBy].toUpperCase().includes(filter.toUpperCase())
       );
+    } else {
+      filtered = items
     }
 
-    return sort(filtered)[this.descOrder ? "desc" : "asc"](
-      (t: any) => t[this.sortBy]
-    );
+    if (sortBy) {
+      return sort(filtered)[this.descOrder ? "desc" : "asc"](
```

Comment:
I think, ```sort``` mutates the original param it's given. Just keep this in mind, shouldn't be an issue when you're doing the ```slice``` I just suggested.

Reply:
added .slice() as mentioned above